lib/internal/process/per_thread.js

```diff
@@ -183,7 +183,7 @@ function wrapProcessMethods(binding) {
   function exit(code) {
     process.off('exit', handleProcessExit);
 
-    if (arguments.length !== 0) {
+    if (code !== undefined && code !== null) {
```

Comment:
This was done on purpose so `process.exit(anyValue)` would be equivalent to `process.exitCode = anyValue; process.exit()`. Why would `null` and `undefined` be treated differently?

Reply:
In the previous versions, there was an implicit condition before `process.exitCode = code;`. It didn't assign the `code` argument to `process.exitCode` when the code was such as `undefined`, `null`, `''`,`NaN`, `false`, and so on. Please see the following code in v19.x.

https://github.com/nodejs/node/blob/4d9af84ea669fb33cef7331faed4064eeedfea2c/lib/internal/process/per_thread.js#L203-L204

In v20.x, the other values except `undefined` or `null` among the above values have been deprecated. So I think that `null` and `undefined` need to be treated as different conditions to meet the previous behavior.

https://github.com/nodejs/node/blob/20af3a63e2653211702a44326ed1f71752b8b551/doc/api/deprecations.md?plain=1#L3203-L3205

Reply:
I'd argue that the new behavior is the correct one, the fact that it was working differently before makes it a breaking change, but that should be OK as it landed as semver-major.

Reply:
Agree that the new behavior is correct and straightforward. Users don't need to know more conditions on how a value passed is treated inside `process.exit()`. It'd be better than this fix if the new one could be acceptable since v20.x.